soroban-env-host/src/native_contract/token/contract.rs

```diff
@@ -159,9 +162,12 @@ impl TokenTrait for Token {
                 write_asset_info(
                     e,
                     AssetInfo::AlphaNum12(AlphaNum12AssetInfo {
-                        asset_code: BytesN::<12>::try_from_val(
+                        asset_code: String::try_from_val(
                             e,
-                            &e.bytes_new_from_slice(&asset12.asset_code.0)?,
+                            &e.string_new_from_slice(
+                                core::str::from_utf8(&asset12.asset_code.0)
```

Comment:
My understanding is that asset codes can contain any data, and are not guaranteed to be utf8. See https://github.com/stellar/stellar-xdr/blob/9e5ddb5be/Stellar-ledger-entries.x#L19-L23.

Are we defining that non-utf8 assets will be unsupported on Soroban?

Reply:
AssetsCodes are limited to alphanumeric data, and this is enforced here - https://github.com/stellar/stellar-core/blob/02232f6061f7accf2036de541d9fb71253e8a699/src/util/types.cpp#L157-L180. ASCII is a subset of UTF-8 so we should be fine here right? We will also add this check to InvokeHostFunctionOp when trying to deploy the SAC.

Reply:
Ah. 👍🏻